org.adempiere.base/src/org/compiere/process/DocumentEngine.java

```diff
@@ -379,10 +379,47 @@ public boolean processIt (String action)
 		}
 		if (ACTION_ReActivate.equals(m_action))
 			return reActivateIt();
-		if (ACTION_Reverse_Accrual.equals(m_action))
-			return reverseAccrualIt();
-		if (ACTION_Reverse_Correct.equals(m_action))
-			return reverseCorrectIt();
+		if (ACTION_Reverse_Accrual.equals(m_action) || ACTION_Reverse_Correct.equals(m_action))
+		{
+			boolean ok = false;
+			if (ACTION_Reverse_Accrual.equals(m_action))
+				ok = reverseAccrualIt();
+			else if (ACTION_Reverse_Correct.equals(m_action))
+				ok = reverseCorrectIt();
+			
+			if (m_document != null && ok)

```

Comment:
a bit more compact to move and add `&& MClient.isClientAccountingImmediate() && (m_document instanceof IDocsPostProcess) && (m_document instanceof PO)` here